src/MCPClient/lib/clientScripts/parse_dataverse_mets.py

```diff
@@ -0,0 +1,265 @@
+                path=path_,
+                checksum=entry.checksum,
+                checksumtype=entry.checksumtype,
+                date=date,
+            )
+
+
+def verify_checksum(job, file_uuid, path, checksum, checksumtype,
+                    event_id=None, date=None):
+    """
+    Verify the checksum of a given file, and create a fixity event.
+
+    :param str file_uuid: UUID of the file to verify
+    :param str path: Path of the file to verify
+    :param str checksum: Checksum to compare against
+    :param str checksumtype: Type of the provided checksum (md5, sha256, etc)
+    :param str event_id: Event ID
+    :param str date: Date of the event
+    """
+    if event_id is None:
+        event_id = str(uuid.uuid4())
+    if date is None:
+        date = timezone.now().isoformat(' ')
+
+    checksumtype = checksumtype.lower()
+    generated_checksum = get_file_checksum(path, checksumtype)
+    event_detail = ('program="python"; '
+                    'module="hashlib.{}()"'.format(checksumtype))
+    if checksum != generated_checksum:
+        job.pyprint('Checksum failed')
+        event_outcome = "Fail"
+        detail_note = 'Dataverse checksum %s verification failed' % checksum
+    else:
+        job.pyprint('Checksum passed')
+        event_outcome = "Pass"
+        detail_note = 'Dataverse checksum %s verified' % checksum
+
+    databaseFunctions.insertIntoEvents(
+        fileUUID=file_uuid,
+        eventIdentifierUUID=event_id,
+        eventType='fixity check',
+        eventDateTime=date,
+        eventDetail=event_detail,
+        eventOutcome=event_outcome,
+        eventOutcomeDetailNote=detail_note,
+    )
+
+
+def parse_(job, unit_path, unit_uuid):
+    """Access the existing METS file and extract and validate its components.
+    """
+    dataverse_mets_path = os.path.join(unit_path, 'metadata', 'METS.xml')
+    mets = metsrw.METSDocument.fromfile(dataverse_mets_path)
+    mapping = get_db_objects(job, mets, unit_uuid)
+    if mapping is None:
+        logger.error(
+            "Exiting. Returning the database objects for our Dataverse "
+            "files has failed.")
+        return 1
+    update_file_use(job, mapping)
+    agent = add_external_agents(job, unit_path)
+    create_derivatives(job, mapping, agent)
+    validate_checksums(job, mapping, unit_path)
+    return 0
+
+
+def parse_dataverse_mets(job):
+    """Extract the arguments provided to the script and call the primary
+    function concerned with parsing Dataverse METS.
+    """
+    try:
+        transfer_dir = job.args[1]
+        transfer_uuid = job.args[2]
+        logger.info("Parse Dataverse METS with dir args: '%s' and transfer "
+                    "uuid: %s", transfer_dir, transfer_uuid)
+        return parse_(job, transfer_dir, transfer_uuid)
+    except IndexError:
+        logger.error(
+            "Problem with the supplied arguments to the function "
+            "len: %s", len(job.args))
+        return 1
```

Comment:
We have a lot of functions in client script doing this same thing, returning exit codes. I'm thinking it'd be much nicer if we separate concerns, e.g. raise an exception instead of returning 1 (`JobContext` will automatically capture and set the status to 1 for us) or just don't return or return `None` which will result in 0.

Reply:
I think your suggestion is neater so will try that out in these scripts. It'll be good to find the consistency in the other scripts over time.

Reply:
When you take a look at the modified PR, I'm not sure if I've addressed all your concerns though, but happy to make changes accordingly.